zerver/lib/test_classes.py

```diff
@@ -2340,6 +2341,13 @@ def fetch_other_streams_subscriber_count(self, stream_ids: set[int]) -> dict[int
             streams=Stream.objects.exclude(id__in=stream_ids)
         )
 
+    def get_dm_group_recipient(self, sender: UserProfile, *other_users: UserProfile) -> Recipient:
+        direct_group_message = get_or_create_direct_message_group(
+            id_list=[sender.id] + [user.id for user in other_users],
+        )
```

Comment:
So reading the code, the question came to me of what `get_or_create_direct_message_group` does if there are duplicate users in the list passed into it... it look like the `group_size` field will be wrong if so. `get_direct_message_group_hash` seems to handle that issue.

I think maybe it's best to just do a new little PR that makes `get_or_create_direct_message_group` just use a `Set` to deduplicate, or maybe better to `assert len(id_list) == len(set(id_list))`, since that can be merged independently of the rest of this train of changes.